### PR TITLE
feat(ref): add VHAL HTTP API documentation

### PR DESCRIPTION
**Pull request by ajanon**
This adds VHAL endpoint documentation to the HTTP API docs based on the current version of the VHAL implementation.

### CODE REVIEW

**Review by morphis — APPROVED** [review]
LGTM

File: ref/anbox-https-api.md [hunk 1]
```diff
@@ -491,3 +494,211 @@ Platform | Field name       | Available since   | JSON type | Access | Descripti
...
+To get the list of available properties and areas, query first the [`1.0/vhal/config` endpoint](#heading--10vhalconfig).
+
+Example return value for `curl -s -X GET --unix-socket /run/user/1000/anbox/sockets/api.unix s/1.0/vhal/0x15600503/0x31 | jq .`:
+
+```bash
+{
+  "metadata": {
+    "value": {
+      "area_id": 49,         # Requested area.
+      "bytes": [],           # Raw bytes value as an 8-bit unsigned integer array.
```
Comment (adglkh) [inline ref/anbox-https-api.md:525]: Same question I left on the MP,  do we need to keep the field when its value is empty? Same for int32_values/int64_values and others.

Comment (ajanon) [reply ref/anbox-https-api.md:525]: I will rework serialization in a follow-up MP.
For now, I prefer to have the documentation reflect what's actually in the code: if we omit the fields in the future, I will update the documentation and remove them (or explain that they may be omitted).

File: ref/anbox-https-api.md [hunk 2]
```diff
@@ -491,3 +494,211 @@ Platform | Field name       | Available since   | JSON type | Access | Descripti
...
+
+```bash
+{
+    "status": "Success",
+    "status_code": 200,
+    "type": "sync"
+}
+```
+
+<a name="heading--10vhalconfig"></a>
```
Comment (keirthana) [inline ref/anbox-https-api.md:595]: These HTML anchors are no longer necessary. The new layout has a page level TOC which is auto-generated. See https://anbox-cloud.io/docs/exp/production for example.

It's best to leave out these anchors as I have a clean up task for them anyway. It breaks headings and navigation in RTD.

File: ref/anbox-https-api.md [hunk 3]
```diff
@@ -491,3 +494,211 @@ Platform | Field name       | Available since   | JSON type | Access | Descripti
...
+`MIXED` property types may have multiple of these values set at the same time, see
+[VHAL property types](https://source.android.com/docs/automotive/vhal/property-configuration#property-types).
+
+Status can be one of the following values, taken from the
+[VhalPropertyStatus](https://cs.android.com/android/platform/superproject/+/android10-release:hardware/interfaces/automotive/vehicle/2.0/types.hal;l=2700-2720)
+enumeration:
+
+|Name|Value|Description|
+|-|-|-|
+|Available|0|Property is available and behaving normally|
```
Comment (keirthana) [inline ref/anbox-https-api.md:555]: Very small nitpick. Better to have punctuation at the end of all sentences or not have have them for all.

**Review by keirthana — CHANGES_REQUESTED** [review]

File: ref/anbox-https-api.md (hunk 1 above)
Comment (adglkh) [reply ref/anbox-https-api.md:525]: That sounds good to me .

**Review by adglkh — APPROVED** [review]
LGTM

**Pull request merged**